app/upgrades/v0.0.12.go

```diff
@@ -0,0 +1,130 @@
+package upgrades
+
+import (
+	"context"
+
+	"cosmossdk.io/math"
+	storetypes "cosmossdk.io/store/types"
+	upgradetypes "cosmossdk.io/x/upgrade/types"
+	cosmosTypes "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/module"
+	"github.com/pokt-network/poktroll/app/keepers"
+	servicetypes "github.com/pokt-network/poktroll/x/service/types"
+	sharedtypes "github.com/pokt-network/poktroll/x/shared/types"
+	suppliertypes "github.com/pokt-network/poktroll/x/supplier/types"
+)
+
+// Upgrade_0_0_12 is the upgrade handler for v0.0.12 upgrade.
+//   - Before: v0.0.11
+//   - After: v0.0.12
+
+// This upgrade introduces a type change to RevSharePercent from float32 to uint64, which is introduced as a separate
+// protobuf field. As a result, we expect existing on-chain data to switch to default value.
+// Investigate the impact of this change on existing on-chain data.
+var Upgrade_0_0_12 = Upgrade{
+	PlanName: "v0.0.12",
+	CreateUpgradeHandler: func(mm *module.Manager,
+		keepers *keepers.Keepers,
+		configurator module.Configurator,
+	) upgradetypes.UpgradeHandler {
+		// Adds new parameters using ignite's config.yml as a reference. Assuming we don't need any other parameters.
+		applyNewParameters := func(ctx context.Context) (err error) {
+			logger := cosmosTypes.UnwrapSDKContext(ctx).Logger()
+			logger.Info("Starting parameter updates for v0.0.12")
+
+			// Add supplier module staking_fee per `config.yml`. The min stake is set to 1000000 upokt, but we avoid
+			// GetParams() to avoid potential protobuf issues and all networks have the same value (no need to read).
+			// Validate with: `poktrolld q supplier params --node=https://testnet-validated-validator-rpc.poktroll.com/`
+			supplierParams := suppliertypes.Params{
+				MinStake: &cosmosTypes.Coin{
+					Denom:  "upokt",
+					Amount: math.NewInt(1000000),
+				},
+				StakingFee: &cosmosTypes.Coin{
+					Denom: "upokt",
+					// TODO_IN_THIS_PR: 100upokt a good value?
+					Amount: math.NewInt(100),
+				},
+			}
+
+			// ALL parameters must be present when setting params.
+			err = keepers.SupplierKeeper.SetParams(ctx, supplierParams)
+			if err != nil {
+				logger.Error("Failed to set supplier params", "error", err)
+				return err
+			}
+			logger.Info("Successfully updated supplier params", "new_params", supplierParams)
+
+			// Add service module `target_num_relays` parameter per `config.yml`.
+			// We don't use `GetParams()` to avoid potential protobuf issues and all networks have the same value (no need to read).
```

Comment:
See my edit above about adding details in "all networks".

1. When I read it, I didn't know i you meant all blockchain networks, all cosmos networks, all versions of shannon (alpha, beta, etc...).

2. Note that this is the case today but it's VERY likely to change soon. Given that we're not in a rush (and nothing is breaking / blocking), now is the time to do it right.

3. These examples will act as copy-pasta references for other people. I'd like to either:
	- Use GetParams if possible
	- Show copy-pasta instructions (docs, comments, etc) on how we ensure that all vars are accounted for.

Reply:
Noted. I'll switch to `GetParams()` everywhere.